src/python/zquantum/qaoa/ansatzes/x_ansatz.py

```diff
@@ -0,0 +1,117 @@
+            number_of_params: number of the parameters that need to be set for the ansatz circuit.
+        """
+
+        super().__init__(number_of_layers)
+        self._cost_hamiltonian = cost_hamiltonian
+
+        assert number_of_layers <= self.number_of_qubits
+
+    @property
+    def number_of_qubits(self):
+        """Returns number of qubits used for the ansatz circuit."""
+        return count_qubits(change_operator_type(self._cost_hamiltonian, QubitOperator))
+
+    @property
+    def number_of_params(self) -> int:
+        """Returns number of parameters in the ansatz."""
+        sum = 0
+        for i in range(1, self.number_of_layers + 1):
+            sum += ncr(self.number_of_qubits, i)
+        return sum
+
+    @overrides
+    def _generate_circuit(self, params: Optional[np.ndarray] = None) -> Circuit:
+        """Returns a parametrizable circuit represention of the ansatz.
+        Args:
+            params: parameters of the circuit.
+        """
+        if params is not None:
+            Warning(
+                "This method retuns a parametrizable circuit, params will be ignored."
+            )
+        circuit = Circuit()
+
+        # Add time evolution layers
+        j = 0
+        for k in range(self.number_of_layers):
+
+            A = combinations(range(0, self.number_of_qubits), k + 1)
+
+            for S in list(A):
+                H_j = QubitOperator(" ".join([f"X{i}" for i in S]))
+
+                circuit += time_evolution(H_j, sympy.Symbol(f"theta_{j}"))
+                j += 1
+
+        return circuit
+
+
+def create_x_qaoa_circuits(
+    hamiltonians: List[QubitOperator], number_of_layers: Union[int, List[int]]
+):
+    """Creates parameterizable quantum circuits based on the X qaoa ansatz for each
+    hamiltonian in the input list using the set number of layers.
+    Args:
+        hamiltonians (List[QubitOperator]): List of hamiltonians for constructing the
+            circuits
+        number_of_layers (Union[int, List[int]]): The k-body depth of the ansatz in the circuit.
+            If an int is passed in, the same k-body depth is used for every ansatz circuit, however,
+            if a list of ints is passed in, the k-body depth used for the hamiltonian at index i of the hamiltonians
+            list is the integer at index i of the number_of_layers list.
+    Returns:
+        List of zquantum.core.circuit.Circuit
+    """
+    if isinstance(number_of_layers, int):
+        number_of_layers = [number_of_layers for _ in range(len(hamiltonians))]
+    number_of_layers_list = number_of_layers
+    assert len(number_of_layers_list) == len(hamiltonians)
+
+    circuitset = []
+    for number_of_layers, hamiltonian in zip(number_of_layers_list, hamiltonians):
+        ansatz = QAOAXAnsatz(number_of_layers, hamiltonian)
+        circuitset.append(ansatz.parametrized_circuit)
+    return circuitset
+
+
+import operator as op
+from functools import reduce
+
+
+def ncr(n, r):
```

Comment:
I have no idea what this function is doing, please rename it to something easier to understand.